ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadManager.swift

```diff
@@ -269,17 +281,23 @@ public class ResourceDownloadManager {
    * `withNotifications` specifies whether or not any of KeymanEngine's `NotificationCenter` notifications should be generated.
    */
   public func downloadPackage<FullID: LanguageResourceFullID>(forFullID fullID: FullID,
+                                                              withKey packageKey: KeymanPackage.Key,
```

Comment:
Not a fan of this, but we don't have terribly much choice at the moment.

Fixing this would require a proper rework of KeymanEngine's published notification API, as it currently requires the actual `LanguageResource` that motivated the download.  Honestly,  the notifications _should_ be reworked to publish either the `KeymanPackage.Key` or the `KeymanPackage` itself.  (Failures would return the `.Key` either way, for obvious reasons.)

Fixing up the notifications like that would be quite a bit more work... and more importantly, it'd be a huge breaking change.  Huge enough to split off into (yet) another PR.